sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1256,14 +1257,24 @@ def _generate_delete_blobs_options(self,
                     if_tags_match_condition=if_tags_match_condition
                 )
 
-            query_parameters, header_parameters = self._generate_delete_blobs_subrequest_options(**options)
+            built_url = "/{}/{}{}".format(quote(container_name), quote(blob_name, safe='/~'), self._query_str)
 
-            req = HttpRequest(
-                "DELETE",
-                "/{}/{}{}".format(quote(container_name), quote(blob_name, safe='/~'), self._query_str),
-                headers=header_parameters
-            )
-            req.format_parameters(query_parameters)
+            # TODO: Shouldn't need to POP these
```

Comment:
Several values need to be popped as I was hitting errors on every single one that I am popping here. Any ideas on if this is the right direction or if we want to refactor / other ways to get the same effect?